packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -0,0 +1,158 @@
+  /**
+   * Button text that will be displayed when the component is collapsed
+   */
+  collapsedButtonText: PropTypes.string,
+  /**
+   * Button text that will be displayed when the component is expanded
+   */
+  expandedButtonText: PropTypes.string,
+  /**
+   * Icon displayed next to the button text
+   */
+  icon: PropTypes.element,
+  /**
+   * The intl object to be injected for translations. Provided by the injectIntl function.
+   */
+  intl: intlShape.isRequired,
+  /**
+   * Sets the animation for the component when it is expanded or collapsed
+   */
+  isAnimated: PropTypes.bool,
+  /**
+   * Sets the initial state of the component to open and expanded
+   */
+  isInitiallyOpen: PropTypes.bool,
+  /**
+   * Callback function triggered when the component is collapsed
+   */
+  onClose: PropTypes.func,
+  /**
+   * Callback function triggered when the component is expanded
+   */
+  onOpen: PropTypes.func,
+};
+
+const defaultProps = {
+  icon: <IconChevronDown />,
+  isAnimated: false,
+  isInitiallyOpen: false,
+};
+
+const animatedStyle = {
+  // fix for the toggle oscillation when setting isAnimated to true
+  transitionDelay: '0.3s',
+  WebkitTransitionDelay: '0.3s', /* Safari */
+};
+
+class ShowHide extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: props.isInitiallyOpen,
+    };
+
+    this.handleOnClick = this.handleOnClick.bind(this);
+  }
+
+  handleOnClick(e) {
+    e.preventDefault();
+
+    // Fire event from toggle handlers
+    if (!this.state.isOpen && this.props.onOpen) {
+      this.props.onOpen();
+    } else if (this.state.isOpen && this.props.onClose) {
+      this.props.onClose();
+    }
+
+    this.setState(prevState => ({
+      isOpen: !prevState.isOpen,
+    }));
+  }
+
+  render() {
+    const {
+      collapsedButtonText,
+      children,
+      expandedButtonText,
+      icon,
+      intl,
+      isAnimated,
+      isInitiallyOpen,
+      onClose,
+      onOpen,
+      preview,
+      ...customProps
+    } = this.props;
+
+    const collapsedText = (collapsedButtonText === undefined) ? intl.formatMessage({ id: 'Terra.showhide.showmore' }) : collapsedButtonText;
+    const expandedText = (expandedButtonText === undefined) ? intl.formatMessage({ id: 'Terra.showhide.hide' }) : expandedButtonText;
+    const buttonText = !this.state.isOpen ? collapsedText : expandedText;
+
+    const showHideClassName = cx([
+      'button',
+      { 'is-open': this.state.isOpen },
+      customProps.className,
+    ]);
+
+    let previewText = '';
+
+    if (preview) {
+      if (preview.slice(-1) === '.') {
+        previewText = `${preview}..`;
+      } else {
+        previewText = `${preview}...`;
+      }
+    }
+
+    const button = (
+      <Button
+        icon={<span className={cx('icon')}>{icon}</span>}
+        aria-expanded={this.state.isOpen}
+        text={buttonText}
+        onClick={this.handleOnClick}
+      />
+    );
```

Comment:
This is up to your preference, do you have any objections to moving the button down into the render function?

```
    return (
      <div {...customProps} className={showHideClassName}>
        {!this.state.isOpen && previewText}
        <Toggle isOpen={this.state.isOpen} isAnimated={this.props.isAnimated} style={isAnimatedStyle}>
          {this.props.children}
        </Toggle>
        <Button
          icon={<span className={cx('icon')}>{icon}</span>}
          aria-expanded={this.state.isOpen}
          text={buttonText}
          onClick={this.handleOnClick}
        />
      </div>
    )
```

Reply:
No objections from me. this change added a little bit more space between the link/button and the text so I had to generate new screenshots for the webdriver tests: https://github.com/cerner/terra-core/pull/1910/commits/f11dc6a8475a61b39ddc9f204be8da50e511fa48

Reply:
Interesting, this change should not have caused any screenshot updates.

Reply:
@StephenEsser  Do you think this was caused from https://github.com/cerner/terra-core/pull/1816?